market-contract/src/main.sw

```diff
@@ -626,6 +622,11 @@ fn cancel_order_internal(order_id: b256) {
         transfer(user, FUEL_ASSET, order.matcher_fee.as_u64());
     }
 
+    // Refund protocol_fee
+    if order.protocol_fee > 0 {
+        transfer(user, FUEL_ASSET, order.protocol_fee);
+    }
+
```

Comment:
Refunding the `protocol_fee` if the order is canceled, following the logic of refunding the `matcher_fee`